src/Umbraco.Cms.ManagementApi/Factories/ExamineIndexViewModelFactory.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Examine;
+using Umbraco.Cms.Core;
+using Umbraco.Cms.Infrastructure.Examine;
+using Umbraco.Cms.ManagementApi.ViewModels.ExamineManagement;
+
+namespace Umbraco.Cms.ManagementApi.Factories;
+
+public class ExamineIndexViewModelFactory : IExamineIndexViewModelFactory
+{
+    private readonly IIndexDiagnosticsFactory _indexDiagnosticsFactory;
+    private readonly IIndexRebuilder _indexRebuilder;
+
+    public ExamineIndexViewModelFactory(IIndexDiagnosticsFactory indexDiagnosticsFactory, IIndexRebuilder indexRebuilder)
+    {
+        _indexDiagnosticsFactory = indexDiagnosticsFactory;
+        _indexRebuilder = indexRebuilder;
+    }
+
+    public ExamineIndexViewModel Create(IIndex index)
+    {
+        var indexName = index.Name;
+
+        IIndexDiagnostics indexDiag = _indexDiagnosticsFactory.Create(index);
+
+        Attempt<string?> isHealthy = indexDiag.IsHealthy();
+
+        var properties = new Dictionary<string, object?>
+        {
+            ["DocumentCount"] = indexDiag.GetDocumentCount(),
+            ["FieldCount"] = indexDiag.GetFieldNames().Count()
+        };
+
+        foreach (KeyValuePair<string, object?> p in indexDiag.Metadata)
+        {
+            properties[p.Key] = p.Value;
+        }
+
+        var indexerModel = new ExamineIndexViewModel
+        {
+            Name = indexName,
```

Comment:
Maybe extract and return the associated searcher name here? It would be nice to have it be an explicit part of the output, so you know what searcher name to use when searching the index ; right now you have to deduct that you need to use the index name as searcher name.